Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +10,232 @@ public static class BalanceStats
+                {
+                    HighestBalanceEver = highestAmountOfPerson;
+                    var currentOne = PersonWithHighestBalance.ToString();
+
+                    if (string.IsNullOrEmpty(currentOne))
+                    {
+                        PersonWithHighestBalance.Append(currentPersonData[0]);
+                    }
+                    else
+                    {
+                        PersonWithHighestBalance.Replace(currentOne, currentPersonData[0]);
+                    }
+
+
+                }
+                else if (highestAmountOfPerson == HighestBalanceEver)
+                {
+                    if (i == peopleAndBalances.Length - 1)
+                    {
+                        PersonWithHighestBalance.Append(" and ");
+                    }
+                    else
+                    {
+                        PersonWithHighestBalance.Append(", ");
+                    }
+
+                    PersonWithHighestBalance.Append(currentPersonData[0]);
+                }
+
+
+            }
+
+            var answer = $"{PersonWithHighestBalance.ToString()} had the most money ever. ¤{HighestBalanceEver}.";
+
+            TextTable.Build(answer, 3);
+
+            return answer;
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+
+            var HighestLossEver = decimal.MinValue;
+            var PersonWithHighestLoss = "";
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                var allAmounts = currentPersonData[1..];
+
+                // calculateLoss
+
+                if (allAmounts.Length <= 1)
+                {
+                    return "N/A.";
+                }
+
+                var lossForCurrentPerson = 0M;
+
+                for (int j = 0; j < allAmounts.Length - 1; j++)
+                {
+                    var amount1 = decimal.Parse(allAmounts[j]);
+                    var amount2 = decimal.Parse(allAmounts[j + 1]);
+                    lossForCurrentPerson = amount1 - amount2;
+                }
+
+
+                //check if loss is greater then the current highest loss 
+                if (lossForCurrentPerson > HighestLossEver)
+                {
+                    HighestLossEver = lossForCurrentPerson;
+                    PersonWithHighestLoss = currentPersonData[0];
+                }
+
+            }
+
+            var answer = $"{PersonWithHighestLoss} lost the most money. -¤{HighestLossEver}.";
+
+            TextTable.Build(answer, 3);
```

Comment:
This should go outside. The function result should be as in the examples. However, the use of this function in Main should be with Text Table. Don't couple both.

Reply:
is deleted in the last version which is just not committed because some tests are still failing